test/integration/rbac_test.go

```diff
@@ -0,0 +1,456 @@
+		Verbs:     []string{"*"},
+		APIGroups: []string{"*"},
+		Resources: []string{"*"},
+	}
+
+	ruleReadPods = v1alpha1.PolicyRule{
+		Verbs:     []string{"list", "get", "watch"},
+		APIGroups: []string{""},
+		Resources: []string{"pods"},
+	}
+
+	ruleWriteJobs = v1alpha1.PolicyRule{
+		Verbs:     []string{"*"},
+		APIGroups: []string{"batch"},
+		Resources: []string{"*"},
+	}
+)
+
+func TestRBAC(t *testing.T) {
+	superUser := "admin"
+
+	tests := []struct {
+		bootstrapRoles bootstrapRoles
+
+		requests []request
+	}{
+		{
+			bootstrapRoles: bootstrapRoles{
+				clusterRoles: []v1alpha1.ClusterRole{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "allow-all"},
+						Rules:      []v1alpha1.PolicyRule{ruleAllowAll},
+					},
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "read-pods"},
+						Rules:      []v1alpha1.PolicyRule{ruleReadPods},
+					},
+				},
+				clusterRoleBindings: []v1alpha1.ClusterRoleBinding{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "read-pods"},
+						Subjects: []v1alpha1.Subject{
+							{Kind: "User", Name: "pod-reader"},
+						},
+						RoleRef: v1.ObjectReference{Kind: "ClusterRole", Name: "read-pods"},
+					},
+				},
+			},
+			requests: []request{
+				{superUser, "GET", "", "pods", "", "", "", http.StatusOK},
+				{superUser, "GET", "", "pods", api.NamespaceDefault, "a", "", http.StatusNotFound},
+				{superUser, "POST", "", "pods", api.NamespaceDefault, "", aPod, http.StatusCreated},
+				{superUser, "GET", "", "pods", api.NamespaceDefault, "a", "", http.StatusOK},
+
+				{"bob", "GET", "", "pods", "", "", "", http.StatusForbidden},
+				{"bob", "GET", "", "pods", api.NamespaceDefault, "a", "", http.StatusForbidden},
+
+				{"pod-reader", "GET", "", "pods", "", "", "", http.StatusOK},
+				{"pod-reader", "POST", "", "pods", api.NamespaceDefault, "", aPod, http.StatusForbidden},
+			},
+		},
+		{
+			bootstrapRoles: bootstrapRoles{
+				clusterRoles: []v1alpha1.ClusterRole{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "write-jobs"},
+						Rules:      []v1alpha1.PolicyRule{ruleWriteJobs},
+					},
+				},
+				clusterRoleBindings: []v1alpha1.ClusterRoleBinding{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "write-jobs"},
+						Subjects:   []v1alpha1.Subject{{Kind: "User", Name: "job-writer"}},
+						RoleRef:    v1.ObjectReference{Kind: "ClusterRole", Name: "write-jobs"},
+					},
+				},
+				roleBindings: []v1alpha1.RoleBinding{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "write-jobs", Namespace: "job-namespace"},
+						Subjects:   []v1alpha1.Subject{{Kind: "User", Name: "job-writer-namespace"}},
+						RoleRef:    v1.ObjectReference{Kind: "ClusterRole", Name: "write-jobs"},
+					},
+				},
+			},
+			requests: []request{
+				// Create the namespace used later in the test
+				{superUser, "POST", "", "namespaces", "", "", jobNamespace, http.StatusCreated},
+
+				{"user-with-no-permissions", "POST", "batch", "jobs", "job-namespace", "", aJob, http.StatusForbidden},
```

Comment:
You are testing denial when the object does not exist.  You should also test denial when the object does exist.